### ISSUE TITLE
Vue router adds trailing slash if base config provided

### ISSUE THREAD

**Issue by tragid** [vuejs/router#1677]
### Version
4.1.6

### Reproduction link
[github.com](https://github.com/tragid/vue-router-slash-bug)

### Steps to reproduce
1. Create vue3 app with router and custom base route:

```
console.log(window.location.href); // http://localhost:8080/en-us

const router = createRouter({
    history: createWebHistory('/en-us'),
    routes,
  });

console.log(window.location.href); // http://localhost:8080/en-us/
```

2. Open main page http://localhost:8080/en-us

### What is expected?
Location remains as http://localhost:8080/en-us

### What is actually happening?
Location changed to http://localhost:8080/en-us/

**Comment by posva** [comment]
This is actually expected and consistent with how the root `/` is handled without a base. It also makes links in general consistent as they **all start with /**. It's worth noting the browser hides the root `/`, you can try yourself:

```js
history.pushState({}, '', '/')
// undefined
location.href
// 'http://127.0.0.1:5173/'
```

I recommend you to setup a redirect in your server and to set `/en-us/` as canonical with the [canonical link](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/rel#attr-canonical).

Is there any technical reason you want the location to remain as `/en-us` instead of `/en-us/`?

**Comment by posva** [comment]
> It looks strange that developer can control trailing slash on specific route, but can't control it for home root with base provided.

It's not something that changes the behavior so I rather have something that just works by default than yet another thing to care about. The trailing slash can be controlled for routes other than the root and when defining a base, the root is consistently at `/base/`

>  Now we have problems with migrating to vue3 and vue-router 4.

What problems? I'm still looking for a non cosmetic reason to see if this should be added back or not

**Comment by tragid** [comment]
Our SEO team asks developers to generate consistent links for each routes (without trailing slashes). What was possible on vue-router3,  not possible on vue-router4 now.
I understand that this is expected behaviour, `/` for home route will consisently be converted to `/en-us/`.
But it would be great that it could be controlled, as it could lead to failing tests during migration

**Comment by posva** [comment]
Upon further research, I found two things:

- Vue Router 3 actually adds the trailing slash as well when navigating to the root **creating an inconsistent behavior** and leading to other SEO problems. Now I remember that I made sure to keep the resolving of the URL consistent when loading the page and navigating
- Regarding SEO: having a trailing slash doesn't matter for SEO as long as you properly configure the canonical link (see link in the first comment) and consistently refer to **only the trailing slash version** of it on your website (which vue-router will do by default when resolving links).

In conclusion, for a good SEO with a defined base:

- Setup a redirect 301 or 302 from `/en-us` to `/en-us/`
- Make sure to only use links to `/en-us/` in your code, or, use the named route: `router.push({ name: 'home' })`

Note that **these two steps are necessary no matter if you want a trailing slash or not** (tweaking the values of course). Therefore, adding an option to set the trailing slash on the root path when a base is defined (quite a specific case) would not help in any way
